### PR TITLE
Upgrade crypto functions to support OTP 23

### PR DESCRIPTION
**Pull request by noahshaw11**
Upgrade crypto functions to support OTP 23 with backwards compatibility.

Deprecation notice: https://erlang.org/documentation/doc-11.2/doc/general_info/deprecations.html#otp-21

References:

- https://erlang.org/doc/man/crypto.html#crypto_one_time-5
- https://docs.huihoo.com/erlang/r15b03-1/lib/crypto-2.2/doc/html/crypto.html#aes_cfb_128_encrypt-3
- http://erlang.org/documentation/doc-9.2/lib/crypto-4.2/doc/html/crypto.html#block_encrypt-4
- http://erlang.org/documentation/doc-9.2/lib/crypto-4.2/doc/html/crypto.html#hmac-3
- https://erlang.org/doc/man/crypto.html#mac-4
- https://erlang.org/doc/man/ssl.html#cipher_suites-2

### CODE REVIEW

File: src/mochiweb_socket.erl [hunk 1]
```diff
@@ -30,7 +30,7 @@ listen(Ssl, Port, Opts, SslOpts) ->
     end.
 
 add_unbroken_ciphers_default(Opts) ->
-    Default = filter_unsecure_cipher_suites(ssl:cipher_suites()),
+    Default = filter_unsecure_cipher_suites(ssl:cipher_suites(default, 'tlsv1.3')),
```
Comment (etrepum) [inline src/mochiweb_socket.erl:33]: I believe this is what is breaking backwards compatibility

**Comment by noahshaw11** [comment]
See https://github.com/apache/couchdb-mochiweb/pull/5#issuecomment-850457992 for an explanation on why a recompiled `rebar` file was provided. This file can be recompiled by the following commands:
```
asdf install erlang R15B03-1
asdf erlang shell R15B03-1
git clone https://github.com/rebar/rebar.git
cd rebar/ && ./bootstrap
```

**Comment by noahshaw11** [comment]
@etrepum Backwards compatibility should be functional now. Thank you for the review!

**Comment by etrepum** [comment]
I've just tagged a release including this update https://github.com/mochi/mochiweb/releases/tag/v2.21.0

**Comment by noahshaw11** [comment]
Great! Thank you!

**Pull request merged**